cfme/cloud/tenant.py

```diff
@@ -194,29 +193,35 @@ def is_displayed(self):
             self.entities.breadcrumb.active_location == 'Tag Assignment')
 
 
-class Tenant(Navigatable):
-    _param_name = 'Tenant'
+class TenantCollection(Navigatable):
+    """Collection object for the :py:class:`cfme.cloud.tenant.Tenant`."""
 
-    def __init__(self, name, provider, appliance=None):
-        """Base class for a Tenant"""
-        self.name = name
-        self.provider = provider
-        Navigatable.__init__(self, appliance=appliance)
+    def create(self, name, provider, applience=None, cancel=False, wait=False):
+        '''Add a cloud Tenant from the UI and return the Tenant object
+
+        Args:
+            name(str): name of the tenant to add
+            provider(obj): provider object on which you want to add the tenant
+            cancel(bool): True if you want to click cancel after filling the tenant form
+            wait(bool): True if you want to wait till the tenant appears on the UI
 
-    def create(self, cancel=False, wait=False):
-        view = navigate_to(self, 'Add')
-        view.form.fill({
-            'cloud_provider': self.provider.name,
-            'name': self.name
+        Returns:
+             an instance of `cfme.cloud.tenant.Tenant`
+        '''
+        page = navigate_to(self, 'Add')
+        page.form.fill({
+            'cloud_provider': provider.name,
+            'name': name
         })
         if cancel:
-            view.form.cancel_button.click()
+            page.form.cancel_button.click()
         else:
-            view.form.save_button.click()
+            page.form.save_button.click()
```

Comment:
**REQUIRED**
What will happen if the form is not filled and `cancel` is `False`?